src/widgets/chart.rs

```diff
@@ -691,50 +690,58 @@ impl<'a> Chart<'a> {
 
     /// Compute the internal layout of the chart given the area. If the area is too small some
     /// elements may be automatically hidden
-    fn layout(&self, area: Rect) -> ChartLayout {
-        let mut layout = ChartLayout::default();
+    fn layout(&self, area: Rect) -> Option<ChartLayout> {
         if area.height == 0 || area.width == 0 {
-            return layout;
+            return None;
         }
         let mut x = area.left();
         let mut y = area.bottom() - 1;
 
+        let mut label_x = None;
         if self.x_axis.labels.is_some() && y > area.top() {
-            layout.label_x = Some(y);
+            label_x = Some(y);
             y -= 1;
         }
 
-        layout.label_y = self.y_axis.labels.as_ref().and(Some(x));
+        let label_y = self.y_axis.labels.as_ref().and(Some(x));
         x += self.max_width_of_labels_left_of_y_axis(area, self.y_axis.labels.is_some());
 
+        let mut axis_x = None;
         if self.x_axis.labels.is_some() && y > area.top() {
-            layout.axis_x = Some(y);
+            axis_x = Some(y);
             y -= 1;
         }
 
+        let mut axis_y = None;
         if self.y_axis.labels.is_some() && x + 1 < area.right() {
-            layout.axis_y = Some(x);
+            axis_y = Some(x);
             x += 1;
         }
 
-        if x < area.right() && y > 1 {
-            layout.graph_area = Rect::new(x, area.top(), area.right() - x, y - area.top() + 1);
+        let graph_width = area.right().saturating_sub(x);
+        let graph_height = y.saturating_sub(area.top()).saturating_add(1);
+        if graph_width == 0 || graph_height == 0 {
+            return None;
```

Comment:
Could this be moved to the top of the method? To trigger the return earlier if the condition is met.

Reply:
At the top is already a check for total area == 0. I am not even sure this one is actually needed as the lines above should be skipped when there isn't enough space.

It might be another idea to just assert in this position as it should never be == 0 with the check above.

Reply:
Nevermind we use `x` and `y` which can be modified just above.

> It might be another idea to just assert in this position as it should never be == 0 with the check above

It could be 0 because of the substraction.

The check at the top is actually the one that could be removed because this is in the render. But it's in another method so might be best to leave it.

Reply:
> Nevermind we use `x` and `y` which can be modified just above.

Yes, but the whole idea of that method is to not do it when there is not enough space.

> The check at the top is actually the one that could be removed[…]

It's basically the early check you asked for? 🤔

Reply:
I switched the check / early return into debug assertions. They should never trigger as the code above should be correct. Debug assertions should be enough as people will test with debug enabled first. I don't really see the point of enabling them on release builds too.

Also, even without this != 0 check the Chart renders just fine without any panic on empty areas. It's just an early abort to save resources.

Reply:
> It's basically the early check you asked for?

Kind of. I just took a more in depth look at the code and saw the `render` method does the same check too.

> I switched the check / early return into debug assertions

That's a good idea. I think we're good with this version of the code.